### PR TITLE
[Core] Minor refactor using `IsActive`

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Minor rafactor using `IsActive`

**🆕 Changelog**

- [Refactor GiD containers](https://github.com/KratosMultiphysics/Kratos/commit/f17460a67cb709ad0a541a36d7665ffd81234a07)
- [Refactor B&S](https://github.com/KratosMultiphysics/Kratos/commit/62e8709f96651d29c205d03c94049780c20eb2e6)
- [Refactor utilities](https://github.com/KratosMultiphysics/Kratos/commit/96df0d140479d817170f030d105e1bf8c346ac17)
- [Refactor processes](https://github.com/KratosMultiphysics/Kratos/commit/17b45ff92088bf5b5b3ccd77577a520385225394)

### CODE REVIEW

**Comment by loumalouomega** [comment]
@roigcarlo any idea why this? (only windows)

~~~sh
     Creating library D:/a/Kratos/Kratos/build/Custom/kratos/Custom/Kratos.lib and object D:/a/Kratos/Kratos/build/Custom/kratos/Custom/Kratos.exp
add_other_utilities_to_python.obj : error LNK2019: unresolved external symbol "public: bool __cdecl Kratos::MasterSlaveConstraint::IsActive(void)const " (?IsActive@MasterSlaveConstraint@Kratos@@QEBA_NXZ) referenced in function "public: __cdecl <lambda_05c4e18abb9c086029edb767fe8acad6>::operator()(class Kratos::MasterSlaveConstraint &)const " 
~~~

**Comment by philbucher** [comment]
> @roigcarlo any idea why this? (only windows)
> 
> ```shell
>      Creating library D:/a/Kratos/Kratos/build/Custom/kratos/Custom/Kratos.lib and object D:/a/Kratos/Kratos/build/Custom/kratos/Custom/Kratos.exp
> add_other_utilities_to_python.obj : error LNK2019: unresolved external symbol "public: bool __cdecl Kratos::MasterSlaveConstraint::IsActive(void)const " (?IsActive@MasterSlaveConstraint@Kratos@@QEBA_NXZ) referenced in function "public: __cdecl <lambda_05c4e18abb9c086029edb767fe8acad6>::operator()(class Kratos::MasterSlaveConstraint &)const " 
> ```

`MasterSlaveConstraint` is missing `KRATOS_API`

**Comment by roigcarlo** [comment]
You are using it now in files that are included by headers in the python interface

**Comment by loumalouomega** [comment]
> You are using it now in files that are included by headers in the python interface

**Comment by loumalouomega** [comment]
Ready @roigcarlo @philbucher

File: kratos/processes/skin_detection_process.cpp [hunk 1]
```diff
@@ -77,11 +77,7 @@ void SkinDetectionProcess<TDim>::GenerateFaceMaps(
         auto it_elem = it_elem_begin + i;
 
         // Detect if the element is active or not. If the user did not make any choice the element is active by default
```
Comment (philbucher) [inline kratos/processes/skin_detection_process.cpp]: please remove those comments, as they are now not applicable any more

File: kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h [hunk 2]
```diff
@@ -1252,14 +1244,9 @@ class ResidualBasedBlockBuilderAndSolver
...
-                //detect if the element is active or not. If the user did not make any choice the element
-                //is active by default
-                bool element_is_active = true;
-                if( (it)->IsDefined(ACTIVE) ) {
-                    element_is_active = (it)->Is(ACTIVE);
-                }
-
-                if(element_is_active) {
+                // Detect if the element is active or not. If the user did not make any choice the element
+                // is active by default
```
Comment (philbucher) [inline kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h]:
```suggestion
```

e.g. here

**Review by philbucher — APPROVED** [review]
much cleaner, thanks!

FYI @KratosMultiphysics/geomechanics I think you also wanted a function to check the active status

**Pull request merged**